File: src/main/java/de/digitalstep/ntlmproxy/NtlmProxy.java
package de.digitalstep.ntlmproxy;

import static java.util.concurrent.Executors.newCachedThreadPool;

import java.net.ServerSocket;
import java.net.Socket;
import java.util.concurrent.ExecutorService;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.google.common.base.Optional;

import de.digitalstep.ntlmproxy.ui.TrayIconBuilder;

public class NtlmProxy {

    private static final Logger log = LoggerFactory.getLogger(NtlmProxy.class);
    private static final ExecutorService EXECUTOR = newCachedThreadPool();

    public static void main(String[] args) throws Exception {
        System.setProperty("java.net.useSystemProxies", "true");

        final Optional<HandlerListener> listener = initUI();

        final int port = Integer.parseInt(args[0]);
        ServerSocket server = new ServerSocket(port);
        log.info("Proxy listening on port {}", port);
        while (true) {
            Socket socket = server.accept();
            EXECUTOR.execute(new Handler(socket, listener));
        }
    }

    private static Optional<HandlerListener> initUI() {
        return new TrayIconBuilder().build();
    }

}

File: src/main/scala/de/digitalstep/ntlmproxy/NtlmProxy.scala
package de.digitalstep.ntlmproxy

import com.google.common.base.Optional
import de.digitalstep.ntlmproxy.ui.TrayIconBuilder
import java.net.ServerSocket
import com.typesafe.scalalogging.slf4j.Logging
import java.util.concurrent.Executors
import de.digitalstep.ntlmproxy.ui.SystemTrayBuilder

object NtlmProxy extends Logging {
	
	val EXECUTOR = Executors.newCachedThreadPool

	def main(args: Array[String]) {
		System.setProperty("java.net.useSystemProxies", "true")

		val listener = SystemTrayBuilder() 
		val port = Integer.parseInt(args(0))
		val server = new ServerSocket(port)
		logger.info(s"Proxy listening on port ${port}")
		while (true) {
			val socket = server.accept()
			EXECUTOR.execute(new Handler(socket, listener))
		}
		
	}

}